examples/custom_shape/stages.py

```diff
@@ -0,0 +1,59 @@
+from locust import HttpUser, TaskSet, task, constant
+from locust import LoadTestShape
+
+
+class UserTasks(TaskSet):
+    @task
+    def get_root(self):
+        self.client.get("/")
+
+
+class WebsiteUser(HttpUser):
+    wait_time = constant(0.5)
+    tasks = [UserTasks]
+
+
+class StagesShape(LoadTestShape):
+    """
+    A simply load test shape class that has different user and hatch_rate at
+    different stages.
+
+    Keyword arguments:
+
+        stages -- A list of dicts, each representing a stage with the following keys:
+            duration -- When this many seconds pass the test is advanced to the next stage
+            users -- Total user count
+            hatch_rate -- Hatch rate
+            stop -- A boolean that can stop that test at a specific stage
```

Comment:
Maybe a None entry could be used to signal the end of the test? Because you'll never need to do (x, y, True) for any other values of x or y than zero, right?

Reply:
Well we need to return at least `(Int, Int)` to edit the running test so you are saying to stop the test return `(None, None)`? Or `(Int, Int, None)`? Or something else?

Reply:
Ah, I kind of conflated two issues: 

I meant that tick() should do `return None` instead of `return (0, 0, True)` (which of course requires modifications in runners.py on line 297 as well.

But I also think that stop_at_end is an unnecessary complication. If you want to run your test forever, cant you just set the last step to use `'duration': 999999` ?

Reply:
> I meant that tick() should do return None instead of return (0, 0, True)

Is that consistent? Wouldn't something like `(Int, Int)`, `(None, None)` or `()` be better?

> But, I also think that stop_at_end is an unnecessary complication

This is just an example of what I imagine someone might want to use or see before they write their own. But I can remove `stop_at_end` it to make it simpler, no worries.

> cant you just set the last step to use 'duration': 9999999

I would say no, that's not very precise. What if you forget it's running and it gets to 10000000 seconds😃

Reply:
Just to reiterate, these are examples. I imagine they would be in the docs or just here for people to find. They aren't to be copy/pasted and used as they are. I just imagined them to help a beginner to see how this class could be used 🙂

Reply:
Sure. Lets not spend too much time on them. On the other hand, if they are just examples then it makes sense to keep them very simplified (and let our users worry about edge cases like how to make it run forever etc).